server/services/image/service.ts

```diff
@@ -0,0 +1,78 @@
+import { Readable } from 'stream';
+
+import fileType from 'file-type';
+import sharp from 'sharp';
+import isSvg from 'is-svg';
+
+import { ServiceClass } from '../../sdk/types/ServiceClass';
+import { IMediaService, ResizeResult } from '../../sdk/types/IMediaService';
+
+export class MediaService extends ServiceClass implements IMediaService {
+	protected name = 'media';
+
+	private imageExts = new Set([
+		'jpg',
+		'png',
+		'gif',
+		'webp',
+		'flif',
+		'cr2',
+		'tif',
+		'bmp',
+		'jxr',
+		'psd',
+		'ico',
+		'bpg',
+		'jp2',
+		'jpm',
+		'jpx',
+		'heic',
+		'cur',
+		'dcm',
+	]);
+
+	resizeFromBuffer(input: Buffer, width: number, height: number, keepType: boolean, blur: boolean, enlarge: boolean, fit?: keyof sharp.FitEnum | undefined): Promise<ResizeResult> {
+		const transformer = sharp(input)
+			.resize({ width, height, fit, withoutEnlargement: !enlarge });
+
+		if (!keepType) {
+			transformer.jpeg();
+		}
+
+		if (blur) {
+			transformer.blur();
+		}
+
+		return transformer.toBuffer({ resolveWithObject: true }).then(({ data, info: { width, height } }) => ({ data, width, height }));
```

Comment:
let me be a bit picky here 🙈 

can we avoid using `when` whenever possible? 😬 IMO the code using `await` is better for read, so it'd look like something like this (in this example, might be better to rename params so doesn't need to rename the result):

```javascript
		const { 
			data, 
			info: { 
				width: widthInfo, 
				height: heightInfo,
			},
		} = await transformer.toBuffer({ resolveWithObject: true });

		return {
			data,
			width: widthInfo,
			height: heightInfo,
		};